### PR TITLE
NAS-116230 / 22.12 / NAS-116230: Remove Core mentions and convert Enterprise

### PR DESCRIPTION
**Pull request by denysbutenko**
Changes:

- `ProductType.Core` has been removed
- `ProductType.Enterprise` has been converted to `ProductType.ScaleEnterprise` and removed
- Some refactoring for getting productType

### CODE REVIEW

File: src/app/modules/common/dialog/about/about-dialog.component.html [hunk 1]
```diff
@@ -2,8 +2,7 @@
   <div class="header">
     <div class="header-overlay">
       <div class="logo-wrapper">
-        <mat-icon *ngIf="systemType === ProductType.Enterprise" svgIcon="truenas_enterprise_logo_full" class="logo" alt="TrueNAS ENTERPRISE logo"></mat-icon>
```
Comment (undsoft) [inline src/app/modules/common/dialog/about/about-dialog.component.html:5]: You may be able to remove the image too.

File: src/app/pages/system/update/update.component.html [hunk 2]
```diff
@@ -124,34 +124,15 @@
   </mat-card-content>
 </mat-card>
 
-<mat-card *ngIf="updates_available && pre_release_train && product_type === ProductType.Core" id="stable-warning-card">
-  <div class="stable-warning"><i class="material-icons" id="update-info">info</i>
-    <strong>{{"This is not a production release, and should only be used for testing." | translate}}</strong>
```
Comment (undsoft) [inline src/app/pages/system/update/update.component.html:129]: This looks useful.

Comment (denysbutenko) [reply src/app/pages/system/update/update.component.html:129]: Is there some case when condition can be true? The same block with other сondition is present below

Comment (undsoft) [reply src/app/pages/system/update/update.component.html:129]: I'm just saying that I think it makes sense to reimplement this for scale (assuming that pre_release_train works correctly).

**Pull request merged**